File: app-flutter/lib/pages/flutter/layout/list/Wrap.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class WrapPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    WrapPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}

class _PageState extends State<WrapPage> {
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
                title:Config.appBar(widget.title),
                centerTitle:true,
            ),
            body: ListView(
                padding:EdgeInsets.all(10.0),
                children: <Widget>[
                    Container(
                        height: 200.0,
                        color: Colors.black26,
                        child: Wrap(
                            spacing:10.0,
                            runSpacing: 10.0,
                            direction: Axis.horizontal,
                            crossAxisAlignment: WrapCrossAlignment.end,
                            runAlignment:  WrapAlignment.center,
                            alignment: WrapAlignment.center,
                            children: <Widget>[
                                Container(width:100.0,height: 50.0,color: Colors.red),
                                Container(width:100.0,height: 50.0,color: Colors.yellow),
                                Container(width:100.0,height: 50.0,color: Colors.blue),
                                Container(width:100.0,height: 50.0,color: Colors.black),
                                Container(width:100.0,height: 50.0,color: Colors.pink),
                                Container(width:100.0,height: 50.0,color: Colors.green)
                            ],
                        ),
                    ),
                ],
            )
        );
    }
}

File: app-react/src/pages/flutter/widget/layout/list/Wrap.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'多元素容器',
	keyword:'Wrap( )',
	type:'Wrap',
	url:'WrapPage',
	data:[
		{ keyword:'direction', type:'Axis', pro:'排列方向，默认水平', code:[ {path:'flutter/widget/layout/list/Wrap/Axis'} ] },
		
		{ deviceTitle:'对齐方式', keyword:'alignment', type:'WrapAlignment', pro:'主轴对齐方式，', code:[ {path:'flutter/widget/layout/list/Wrap/WrapAlignment'} ] },
		{ keyword:'runAlignment', type:'WrapAlignment', pro:'副轴对齐方式', code:[ {path:'flutter/widget/layout/list/Wrap/WrapAlignment'} ] },
		{ keyword:'crossAxisAlignment', type:'WrapCrossAlignment', pro:'横轴对齐方式', code:[ {path:'flutter/widget/layout/list/Wrap/WrapCrossAlignment'} ] },
		
		{ deviceTitle:'间距', keyword:'spacing', type:'double', pro:'水平间距' },
		{ keyword:'runSpacing', type:'double', pro:'垂直间距' },
		
		{ deviceTitle:'其它',keyword:'verticalDirection', type:'VerticalDirection', pro:'排列摆放顺序', code:[ {path:'flutter/widget/layout/list/Wrap/VerticalDirection'} ] },
		{ keyword:'textDirection', type:'TextDirection', pro:'排列顺序', code:[ {path:'flutter/widget/text/TextDirection'} ] },
		{ keyword:'children', type:'List<Widget>', pro:'列表控件', code:[ {path:'flutter/public/widget-children'} ]  }
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }